code/modules/mob/living/say.dm

```diff
@@ -289,8 +289,24 @@ proc/get_radio_key_from_channel(var/channel)
 	return 1
 
 /mob/living/proc/say_signlang(var/message, var/verb="gestures", var/datum/language/language)
-	for(var/mob/O in viewers(src, null))
-		O.hear_signlang(message, verb, language, src)
+	if(restrained())
+		to_chat(src, "<span class='danger'>You're restrained and cannot sign!</span>")
+		return
+
+	if(ishuman(src))
+		var/mob/living/carbon/human/C = src
+		var/obj/item/organ/external/rhand = C.bodyparts_by_name["r_hand"]
+		var/obj/item/organ/external/lhand = C.bodyparts_by_name["l_hand"]
+		if((!rhand || !rhand.is_usable() || rhand.is_broken()) || (!lhand || !lhand.is_usable() || lhand.is_broken()))
+			to_chat(src, "<span class='warning'>You try to use your hand to sign, but you can't!</span>")
+			return
+
+	for(var/mob/M in get_mobs_in_view(7, src))
+		if(M.see_invisible < invisibility)

```

Comment:
Under what circumstances would an invisible mob be trying to communicate using sign language?

Reply:
Does we still have that cloaking device? The one that makes you completely invisible? I know SS13 used to have something like that. I abused the shit out of it along with shield.

Also any mob (read: antag) who can phase out of reality on command

Reply:
I included that check to be consistent with how the visible_message proc worked, as that was the recommended way to output visible messages, but I didn't use it because it doesn't do any language checks because it's usually outputting descriptions of actions seen, and opted to use the already existing "say_signlang" proc.

I believe the intention is that if the person signing is invisible, only those with the "see_invisible" flag will see them signing. Think there's a wizard spell that allows seeing invisible?